### PR TITLE
OCT-1369: Closing Calculate Reward modal should cancel the API call to /rewards/estimated_budget if response is not recieved

### PR DESCRIPTION
**Pull request by jmikolajczyk**

### CODE REVIEW

File: client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx [hunk 1]
```diff
@@ -38,44 +33,29 @@ const EarnRewardsCalculator: FC = () => {
...
-        setEstimatedRewards(undefined);
-        setIsFetching(false);
-        return;
-      }
-
-      if (isFetchingRewards) {
-        clientReactQuery.cancelQueries({ queryKey: [ROOTS.calculateRewards] });
-      }
-
+  const fetchEstimatedRewardsDebounced2 = useCallback(
```
Comment (jmikolajczyk) [inline client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx:44]: `fetchEstimatedRewardsDebounced`

File: client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx [hunk 2]
```diff
@@ -38,44 +33,29 @@ const EarnRewardsCalculator: FC = () => {
...
+      mutateAsyncRewardsCalculator({ amountGlm: amountGlmWEI, numberOfDays: numberOfDaysNumber });
     }, 300),
     [],
   );
 
   const formik = useFormik<FormFields>({
     initialValues: formInitialValues,
-    onSubmit: values => fetchEstimatedRewardsDebounced(values.valueCrypto, values.days),
+    onSubmit: values =>
+      fetchEstimatedRewardsDebounced2({ amountGlm: values.valueCrypto, numberOfDays: values.days }),
```
Comment (jmikolajczyk) [inline client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx:58]: `fetchEstimatedRewardsDebounced`

File: client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx [hunk 3]
```diff
@@ -99,14 +79,24 @@ const EarnRewardsCalculator: FC = () => {
 
   useEffect(() => {
     formik.validateForm().then(() => {
-      fetchEstimatedRewardsDebounced(formik.values.valueCrypto, formik.values.days);
+      fetchEstimatedRewardsDebounced2({
```
Comment (jmikolajczyk) [inline client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx:82]: `fetchEstimatedRewardsDebounced`

File: client/src/hooks/mutations/useCalculateRewards.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      abortControllerRef.current = new AbortController();
+      return apiPostCalculateRewards(amountGlm, numberOfDays, abortControllerRef.current.signal);
+    },
+    ...options,
+  });
+
+  const reset = useCallback(() => {
+    abortControllerRef.current?.abort();
+    mutation.reset();
+  }, [abortControllerRef, mutation]);
```
Comment (jmikolajczyk) [inline client/src/hooks/mutations/useCalculateRewards.ts:27]: deps array can be empty in this case.

File: client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx (hunk 2 above)
Comment (aziolek) [reply client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx]: Done.

File: client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx (hunk 3 above)
Comment (aziolek) [reply client/src/components/Earn/EarnRewardsCalculator/EarnRewardsCalculator.tsx]: Done.

File: client/src/hooks/mutations/useCalculateRewards.ts (hunk 4 above)
Comment (aziolek) [reply client/src/hooks/mutations/useCalculateRewards.ts:27]: Any harm in having that? It won't redeclare and linter is happy.

Comment (jmikolajczyk) [reply client/src/hooks/mutations/useCalculateRewards.ts:27]: reference doesn't call rendering, so it is unnecessary, mutation is an object, which means it calls a new reset method every time.

Comment (aziolek) [reply client/src/hooks/mutations/useCalculateRewards.ts]: Done. Removed from `useAllocateSimulate` too.

**Review by aziolek — APPROVED** [review]

**Pull request merged**